kura/org.eclipse.kura.linux.net/src/main/java/org/eclipse/kura/linux/net/wifi/WpaSupplicantManager.java

```diff
@@ -161,28 +142,53 @@ public static boolean isTempRunning() throws KuraException {
         }
     }
 
-    /**
-     * Stops all instances of wpa_supplicant
-     *
-     * @throws Exception
-     */
+	/**
+	 * Stops all instances of wpa_supplicant
+	 * 
+	 * @throws Exception
+	 */
     public static void stop(String ifaceName) throws KuraException {
         try {
-            // kill wpa_supplicant
-            s_logger.debug("stopping wpa_supplicant");
-            String cmd = formSupplicantStopCommand(ifaceName);
-            if (cmd != null && !cmd.isEmpty()) {
-                LinuxProcessUtil.start(cmd);
-                if (ifaceName != null) {
-                    LinuxNetworkUtil.disableInterface(ifaceName);
+            if (!s_isIntelEdison) {
+                int pid = getPid(ifaceName);
+                if (pid >= 0) {
+                    s_logger.info("stopping wpa_suplicant for the {} interface, pid={}", ifaceName, pid);
+
+                    boolean exists = LinuxProcessUtil.stop(pid);
```

Comment:
this part of code is the same as in HostapdManager.java, but also in PppLinux.disconnect
I think that maybe is better to have all this code centralized in LinuxProcessUtil or other utility class.